src/module/iotjs_module_udp.cpp

```diff
@@ -307,52 +282,117 @@ JHANDLER_FUNCTION(Close) {
 }
 
 
-JHANDLER_FUNCTION(GetSockeName) {
-  IOTJS_ASSERT(!"Not implemented");
+GetSockNameFunction(UdpWrap, udp_handle, uv_udp_getsockname);
 
-  iotjs_jhandler_return_null(jhandler);
+JHANDLER_FUNCTION(GetSockeName) {
+  DoGetSockName(jhandler);
 }
 
+#define X(fn)                                                                 \
+  JHANDLER_CHECK_THIS(object);                                                \
+  JHANDLER_CHECK_ARGS_1(number);                                              \
+  \
+  UdpWrap* udp_wrap = UdpWrap::FromJObject(iotjs_jhandler_get_this(jhandler));\
+  IOTJS_ASSERT(udp_wrap != NULL);                                             \
+  \
+  int flag = JHANDLER_GET_ARG(0, number);                                     \
+  int err = fn(udp_wrap->udp_handle(), flag);                                 \
+  \
+  iotjs_jhandler_return_number(jhandler, err);
+
 
 JHANDLER_FUNCTION(SetBroadcast) {
+#if !defined(__NUTTX__)
+  X(uv_udp_set_broadcast);
+#else
   IOTJS_ASSERT(!"Not implemented");
 
   iotjs_jhandler_return_null(jhandler);
+#endif
 }
 
 
 JHANDLER_FUNCTION(SetTTL) {
+#if !defined(__NUTTX__)
+  X(uv_udp_set_ttl);
+#else
   IOTJS_ASSERT(!"Not implemented");
 
   iotjs_jhandler_return_null(jhandler);
+#endif
 }
 
 
 JHANDLER_FUNCTION(SetMulticastTTL) {
+#if !defined(__NUTTX__)
+  X(uv_udp_set_multicast_ttl);
+#else
   IOTJS_ASSERT(!"Not implemented");
 
   iotjs_jhandler_return_null(jhandler);
+#endif
 }
 
 
 JHANDLER_FUNCTION(SetMulticastLoopback) {
+#if !defined(__NUTTX__)
+  X(uv_udp_set_multicast_loop);
+#else
   IOTJS_ASSERT(!"Not implemented");
 
   iotjs_jhandler_return_null(jhandler);
+#endif
 }
 
 
-JHANDLER_FUNCTION(AddMembership) {
+void SetMembership(iotjs_jhandler_t* jhandler, uv_membership membership) {
+#if !defined(__NUTTX__)
```

Comment:
IMHO, it is good to guard with more exceptional and self-explaining condition. And this API is supposed to work with os where running libuv fine, so the `__linux__` isn't enough.  and the same things happen when we start to support `darwin` or `window`.  So i am wondering what other's opinion are